mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js

```diff
@@ -179,10 +176,26 @@ export class ExperimentRunsTableMultiColumnView2 extends React.Component {
         },
         {
           headerName: ATTRIBUTE_COLUMN_LABELS.DATE,
-          field: 'startTime',
+          field: 'runDateInfo',
           pinned: 'left',
           initialWidth: 150,
           cellRenderer: 'dateCellRenderer',
+          cellRendererParams: {
+            onExpand: onExpand,
+          },
+          equals: (dateInfo1, dateInfo2) => {
+            return (
+              dateInfo1.referenceTime === dateInfo2.referenceTime &&
+              dateInfo1.startTime === dateInfo2.startTime &&
+              dateInfo1.experimentId === dateInfo2.experimentId &&
+              dateInfo1.runUuid === dateInfo2.runUuid &&
+              dateInfo1.runStatus === dateInfo2.runStatus &&
+              dateInfo1.isParent === dateInfo2.isParent &&
+              dateInfo1.hasExpander === dateInfo2.hasExpander &&
+              dateInfo1.expanderOpen === dateInfo2.expanderOpen &&
+              _.isEqual(dateInfo1.childrenIds, dateInfo2.childrenIds)
+            );
```

Comment:
Why do we need to check the equals here. Where is this function evoked from and how do we use it?

Reply:
It isn't used directly within mlflow code, but is used by ag-grid to decide whether a cell needs to be re-rendered. After the data is updated, ag-grid can decide not to re-render a cell if the row id is the same and the cell values are considered to be equal. If we didn't implement this then a lot of cells would unnecessarily be re-rendered because by default reference equality is used.

This wasn't needed previously when row ids were assigned by ag-grid as a new set of ids is assigned after data is updated.

Reply:
Since we're using _.equals here already, what's stopping us from doing `_.isEqual(dateInfo1, dateInfo2)`?

Reply:
I was concerned there might be more performance overhead of using `_.isEqual` but I haven't tested that to verify. Using `_.isEqual(dateInfo1, dateInfo2)` would definitely simplify things so I'll switch to that if I don't see a big performance difference.

Reply:
Yeah using `_.isEqual` for the top-level comparisons is slower but it's pretty insignificant compared to the time everything else takes. Eg. when loading 100 more rows with 1000 already loaded I get 25 ms in the value comparisons with the current code and then 100 ms when changing all the comparisons to just use `_.isEqual`, but the whole loading operation takes multiple seconds so I'll make this change.